### ISSUE TITLE
[java] GuardLogStatement reported when exception is passed via toString()

### ISSUE THREAD

**Issue by Emilyaxe** [pmd/pmd#6630]
**Affects PMD Version:** 7.23.0

**Rule:** GuardLogStatement — https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#guardlogstatement

**Description:**

PMD incorrectly reports `GuardLogStatement` on SLF4J logger calls when the argument expression is `e.toString()`. This is a false positive. The expression is evaluated only once at the call site, and SLF4J does not re-evaluate it but directly uses the resulting string for formatting.

**Code Sample demonstrating the issue:**

```java
import java.io.IOException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Main {

    private static final Logger logger = LoggerFactory.getLogger(Main.class);

    public static void main(String[] args) {
        try {
            throw new IOException("simulated IO error");
        } catch (IOException e) {
            //  GuardLogStatement shouldn't report on all three lines (lines11-13)
            logger.warn("Failed to process input: {}", e.toString());
            logger.error("Critical failure: {}", e.toString());
            logger.debug("Debug info: {}", e.toString());
        }
    }
}
```

Actual outcome:** PMD reports a violation at lines 11-13 , but that's wrong. That's a false positive..

**Running PMD through:** CLI

**Comment by zbynek** [comment]
Those lines should be flagged because they are calling the `toString` method before the logging framework decides if the message will be logged. See also https://www.slf4j.org/faq.html#string_contents

The correct ways to log the exception should be
```
logger.warn("Failed to process input:", e); // prints the whole stacktrace
logger.warn("Failed to process input: {}", (Object) e); // prints only the class + message
```
